evproc/processor.py

```diff
@@ -308,11 +311,16 @@ def process(self, ev):
                   ``StopProcessing`` exception raised by a processor.
         """
 
-        # Walk through all the processors and process the event
-        for proc in self._get_procs(ev):
-            try:
-                proc(ev)
-            except exc.StopProcessing as ex:
-                return ex.retval
+        event_deque = collections.deque([ev])
+        while event_deque:
+            ev = event_deque.popleft()
+            # Walk through all the processors and process the event
+            for proc in self._get_procs(ev):
+                try:
+                    additional_events = proc(ev)
+                except exc.StopProcessing as ex:
+                    return ex.retval
```

Comment:
If there are no other events in the queue (which there almost certainly won't be with existing code), then the behavior is unchanged, whether or not `retval` is explicitly set in `StopProcessing`.

Anyway, up to you.